### PR TITLE
TRM-32007: Improve RequestConverter to SolrRequest

### PR DESCRIPTION
**Pull request by LeonardoGonzales**

### CODE REVIEW

**Comment by LeonardoGonzales** [comment]
This code change also includes the solution of this Bug: [TRM-31512](https://www.ebi.ac.uk/panda/jira/browse/TRM-31512)

File: uniprotkb-common/src/test/java/org/uniprot/api/uniprotkb/common/service/uniprotkb/UniProtEntryServiceTest.java [hunk 1]
```diff
@@ -139,11 +136,10 @@ void addIsoFormFalseFilterOnlyIfQueryHasNoAccessionValue() {
                 entryService.createSearchSolrRequest(verifyFailingRequest);
         assertNotNull(verifyFailingSolrRequestCase);
         assertTrue(!verifyFailingSolrRequestCase.getFilterQueries().isEmpty());
-    }
+    }*/
```
Comment (ahmadshadab) [inline uniprotkb-common/src/test/java/org/uniprot/api/uniprotkb/common/service/uniprotkb/UniProtEntryServiceTest.java:140]: Do we not need this test which is commented?

**Review by ahmadshadab — COMMENTED** [review]
Very nice code clean up. thank you.

File: common-rest/src/test/java/org/uniprot/api/rest/service/BasicSearchServiceTest.java [hunk 2]
```diff
@@ -119,254 +94,6 @@ void getEntity_conversionReturningNonNullSucceeds() {
...
-        request.setSort("accession asc");
-
-        SolrRequest solrRequest = service.createSearchSolrRequest(request);
-        assertNotNull(solrRequest);
-
-        // then
-        assertNotNull(solrRequest.getSorts());
-        assertEquals(2, solrRequest.getSorts().size());
-    }
-
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply common-rest/src/test/java/org/uniprot/api/rest/service/BasicSearchServiceTest.java:369]: These are for the Download method that we were not using.

File: aa-rest/src/main/resources/application.properties [hunk 3]
```diff
@@ -7,7 +7,8 @@ springdoc.writer-with-default-pretty-printer=true
 springdoc.pre-loading-enabled=false
 openapi.docs.server=https://rest.uniprot.org
 spring.jackson.default-property-inclusion=non_null
-search.default.page.size=25
+search.request.converter.default-page-size=25
+search.request.converter.default-solr-batch-size=100
 spring.mvc.pathmatch.matching-strategy=ant_path_matcher
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply aa-rest/src/main/resources/application.properties:13]: https://www.ebi.ac.uk/panda/jira/browse/TRM-32033

File: common-rest/src/main/java/org/uniprot/api/common/repository/stream/store/uniprotkb/TaxonomyLineageServiceImpl.java [hunk 4]
```diff
@@ -40,12 +43,7 @@ public class TaxonomyLineageServiceImpl extends BasicSearchService<TaxonomyDocum
...
     @Autowired
     public TaxonomyLineageServiceImpl(TaxonomyLineageRepository taxRepo) {
-        super(
-                taxRepo,
-                new TaxonomyEntryConverter(),
-                new TaxonomySortClause(),
-                getQueryBoosts(),
-                null);
+        super(taxRepo, new TaxonomyEntryConverter(), getLineageRequestConverter());
         this.searchFieldConfig =
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply common-rest/src/main/java/org/uniprot/api/common/repository/stream/store/uniprotkb/TaxonomyLineageServiceImpl.java:47]: It was like this before, I did not want to change it.

File: uniprotkb-common/src/main/java/org/uniprot/api/uniprotkb/common/service/request/UniProtKBRequestConverter.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package org.uniprot.api.uniprotkb.common.service.request;
+
+import org.uniprot.api.common.repository.search.SolrRequest;
+import org.uniprot.api.rest.service.request.RequestConverter;
+
+public interface UniProtKBRequestConverter extends RequestConverter {
+    SolrRequest createProteinIdSolrRequest(String proteinId);
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply uniprotkb-common/src/main/java/org/uniprot/api/uniprotkb/common/service/request/UniProtKBRequestConverter.java:7]: We already have a class SolrRequestConverter

File: common-rest/src/test/java/org/uniprot/api/common/repository/search/SolrRequestConverterTest.java [hunk 6]
```diff
@@ -187,18 +187,18 @@ void doNotCreateQueryBoostsAndHighlightForPageSizeZero() {
...
-                                            .queryFields("field1 field2")
-                                            .build())
+                            /*                            .queryConfig(
+                            SolrQueryConfig.builder()
+                                    .addBoost("field1:value3^3")
+                                    .boostFunctions("f1,f2")
+                                    .highlightFields("h1,h2")
+                                    .queryFields("field1 field2")
+                                    .build())*/
                             .queryField("field1 field2")
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply common-rest/src/test/java/org/uniprot/api/common/repository/search/SolrRequestConverterTest.java:197]: thanks, fixed it!

File: uniprotkb-common/src/main/java/org/uniprot/api/uniprotkb/common/service/request/UniProtKBRequestConverterImpl.java [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+        UniProtKBStreamRequest uniProtRequest = (UniProtKBStreamRequest) request;
+        SolrRequest.SolrRequestBuilder builder =
+                basicConverter.createStreamSolrRequest(uniProtRequest);
+
+        updateIsoformsFilterQuery(builder, uniProtRequest);
+
+        if (isSearchAll(uniProtRequest)) {
+            builder.query(getQueryFieldName(ACTIVE) + ":" + true);
+        }
+        builder.largeSolrStreamRestricted(uniProtRequest.isLargeSolrStreamRestricted());
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply uniprotkb-common/src/main/java/org/uniprot/api/uniprotkb/common/service/request/UniProtKBRequestConverterImpl.java:93]: It is easier to read the code this way.

File: uniprotkb-common/src/test/java/org/uniprot/api/uniprotkb/common/service/uniprotkb/UniProtEntryServiceTest.java (hunk 1 above)
Comment (LeonardoGonzales) [reply uniprotkb-common/src/test/java/org/uniprot/api/uniprotkb/common/service/uniprotkb/UniProtEntryServiceTest.java:140]: These tests are covered as unit test in the new RequestConverterTest and  RequestConverterUtilsTest classes

**Review by ahmadshadab — APPROVED** [review]
It is better if you make the k8s changes and deploy rest with docker image created from your private branch.

I have requested Supun to also take a look.

**Pull request merged**